test/clj/harja/palvelin/palvelut/hallinta/urakoiden_lyhytnimet_test.clj

```diff
@@ -0,0 +1,64 @@
+(ns harja.palvelin.palvelut.hallinta.urakoiden-lyhytnimet-test
+  (:require [clojure.test :refer :all]
+            [harja.palvelin.palvelut.hallinta.urakoiden-lyhytnimet :as urakoiden-lyhytnimet]
+            [com.stuartsierra.component :as component]
+            [harja.palvelin.komponentit.tietokanta :as tietokanta]
+            [harja
+             [testi :refer :all]]))
+
+(defn jarjestelma-fixture [testit]
+  (alter-var-root #'jarjestelma
+    (fn [_]
+      (component/start
+        (component/system-map
+          :db (tietokanta/luo-tietokanta testitietokanta)
+          :http-palvelin (testi-http-palvelin)
+          :lyhytnimien-hallinta (component/using
+                                  (urakoiden-lyhytnimet/->UrakkaLyhytnimienHallinta)
+                                  [:db :http-palvelin])))))
+  (testit)
+  (alter-var-root #'jarjestelma component/stop))
+
+(use-fixtures :each (compose-fixtures
+                      jarjestelma-fixture
+                      urakkatieto-fixture))
+
+(defn- hakuparametrit-kaikki []
+  {:urakkatyyppi :kaikki :vain-puuttuvat false :urakan-tila nil})
+
+(def aktiivisen-oulun-id 26) ; kovakoodaamisen sijaan pitäisikö hakea arvo?
+
+(defn- tallenna-parametrit []
+  {:urakat [{:id aktiivisen-oulun-id, :nimi "Aktiivinen Oulu Testi", :lyhyt_nimi "test123"}]
+   :haku-parametrit (hakuparametrit-kaikki)})
+
+(deftest paivita-lyhytnimet
+  (let [
+        tulos (kutsu-palvelua (:http-palvelin jarjestelma)
+                :hae-urakoiden-nimet +kayttaja-jvh+
+                (hakuparametrit-kaikki))
+        vaihdettu-nimi (kutsu-palvelua (:http-palvelin jarjestelma)
+                         :tallenna-urakoiden-lyhytnimet +kayttaja-jvh+
+                         (tallenna-parametrit))
+        ]
+    (is (:lyhyt_nimi (first (filter (comp #{aktiivisen-oulun-id} :id) tulos))) "Oulun lyhyt nimi")
```

Comment:
Tästä taitaa puuttua = vertailu, jolloin is funktio palauttaa aina true. Asia on helppo testata niin, että muuttaa arvoja sellaiseksi, ettei se voi toimia.

En ole tästä kyllä ihan varma, kun en ole ajanut tätä koodia. Mutta olettaisin, että tämän voisi toimia paremmin näin:
(is (= (:lyhyt_nimi (first (filter (comp #{aktiivisen-oulun-id} :id) tulos))) "Oulun lyhyt nimi"))

Reply:
ei toimi nyt, pitää korjata